presto-generation-processor/src/test/java/com/facebook/presto/tests/QuestionableTests.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.tests;
+
+import com.facebook.presto.framework.ConnectorTestFramework;
+import com.facebook.presto.test.ConnectorTest;
+import org.testng.annotations.Test;
+
+import static com.facebook.presto.test.ConnectorFeature.READ_DATA;
+
+/*
+ * It seems like this might be a questionable way to lay out your tests, but
+ * if you choose to do it, it works. Note that the resulting generated classes
+ * get flattened out into individual files, so not only must they be static,
+ * they also can't refer to static members of the enclosing class.
```

Comment:
Why to support such a case? Even "mentally" -- i mean if it works just because it works, with no additional effort, why not ignore this fact and go without documenting it?